doc/internal/man3/OPTIONS.pod

```diff
@@ -0,0 +1,301 @@
+=pod
+
+=head1 NAME
+
+OPTIONS, OPT_PAIR,
+opt_progname, opt_appname, opt_getprog, opt_init, opt_format,
+opt_int, opt_long, opt_imax, opt_umax, opt_ulong, opt_pair,
+opt_string, opt_cipher, opt_md, opt_next, opt_arg, opt_flag, opt_unknown,
+opt_num_rest, opt_rest, opt_help, opt_isdir
+- Option parsing for commands and tests
+
+=head1 SYNOPSIS
+
```

Comment:
The following symbols are not (yet) documented:
```
const char *opt_path_end(const char *filename);
void opt_begin(void);
const char *format2str(int format);
int opt_int_arg(void);
char *opt_name(void);
int opt_verify(int i, X509_VERIFY_PARAM *vpm);
int opt_rand(int i);
int opt_provider(int i);
void opt_print(const OPTIONS * opt, int doingparams, int width);
int opt_format_error(const char *s, unsigned long flags);
void print_format_error(int format, unsigned long flags);
int opt_printf_stderr(const char *fmt, ...);
```

Reply:
For another PR for someone else to do.  What I wrote is good enough for a new openssl developer to write basic options.  (And `opt_int_arg` should crawl away to somewhere else :)

Reply:
Fine by me - the given coverage of functions should already be very helpful.